### ISSUE TITLE
Toolbar disappearing on map

### ISSUE THREAD

**Issue by jessica-azgs** [StraboSpot/strabo-mobile#83]
When switching from the map view to another view and then back to the map the toolbar usually disappears. It can be brought back by clicking the world/airplane icon. I think there is a problem with the drawing order. This happens on iOS most often and never through the emulator in Chrome.

**Comment by ginxwar** [comment]
Zach is running Chrome 32 and is also experiencing this bug.  I'm running Chrome 41 and Firefox 38 and the toolbar never disappears.  After Zach updated his Chrome, the toolbar works fine.

The toolbar disappears on iOS devices because the web engine underneath is Safari, and I can reproduce the bug in Safari 8.0.4 (10600.4.10.7).

I suspect that this may be a Safari bug and it hasn't been fixed yet.  We certainly can't wait for Apple to fix this so we may need to reposition the toolbar such that it never overlaps the map.  @jessica-azgs, what do you think about redoing the controls and dedicating permanent real estate adjacent to the map div?

I know this isn't an elegant solution.

**Comment by jessica-azgs** [comment]
@ginxwar Since switching to airplane mode fixes it and in the switch to airplane mode `map.render();` is called can't we just call that every time the map page is opened?

**Comment by ginxwar** [comment]
Just tried that but it didn't work.  You'd think that ```map.render()``` would clear the cache.  Also tried ```map.renderSync()``` to no avail.

I also tried to force toggle on/off airplaneMode programmatically during runtime but that didn't work either.  I suspect that doing this competes with the $scope digest cycle and the redraw/paint doesn't actually occur.

**Comment by ginxwar** [comment]
I rolled back to many previous versions, going all the way back to early December.  All previous versions had this issue.

I looked closer.  The toolbar is actually still there, it's just not rendered. In other words, you could tap the zoom + and - buttons if you knew they were there and the map zooms accordingly.  Same thing with drawing tools.

I'm starting to think this is some strange CSS bug.  Clearly the object is accessible and able to receive click/touch events.

**Comment by smrazgs** [comment]
@ginxwar @jessica-azgs so where do we stand on this one-- sounds like a back-burner item since the actual bug is unclear and its platform specific?

**Comment by ginxwar** [comment]
This bug has appeared since day one of this project when I regressed the codebase and ran this against Safari.  Something within Safari's CSS rendering engine is likely causing this issue.  Although it is platform specific, this is the engine that powers all iOS devices.

Not sure how to solve this one at the moment...  Likely a backburner item for now.

**Comment by ginxwar** [comment]
I accidentally found a workaround.  On my iPad mini 2, if I pinch zoom-out, the toolbar comes back with full functionality.  Kinda weird...

**Comment by jessica-azgs** [comment]
Yeah, does it for me too. It didn't use to do that though. I was always pressing the geolocate button on and off to get the toolbar back (or going to another page and back). It's not quite as bad now that the pinch-zoom works, but still dumb and wish there was an actually fix.
